### ISSUE TITLE
Automatically remove unneeded dependencies

### ISSUE THREAD

**Issue by endigma** [packwiz/packwiz#97]
this would require some sort of dependency graph store, but would be very useful

**Comment by Den4enko** [comment]
It would be cool if it was implemented as in apt on debian(mark all automatically installed mods as "dependent" and in the mod.pw.toml themselves to write which mods are needed to work.

**Comment by endigma** [comment]
This could potentially cause issues with cross-source deps, i.e. curseforge mod dependency satisfied by a modrinth sourced mod

**Comment by comp500** [comment]
Yeah, I'll need to read loader metadata (Forge/Fabric/Quilt modid) for this to work well, which I want to do anyway to prevent installing duplicate dependencies from CF/Modrinth

**Comment by endigma** [comment]
Have you considered building out a nix-like dep graph?

**Comment by comp500** [comment]
Not really sure what you mean by that... this would just need metadata files to specify which other files they depend on, along with a flag to specify which files are installed as dependencies. Then whenever a file is removed, make a list of all files not installed as dependencies and iteratively add the dependencies of the files in that list; the remaining files are dependencies that are no longer necessary.

**Comment by fuzzyweapon** [comment]
my only guess is they mean something like an directed acyclic graph.  It's just a way of modeling dependencies (see something like Gradle, which also uses a DAG to model for task running).
